pyroute2/dhcp/client.py

```diff
@@ -211,7 +211,7 @@ async def _expire_lease(self):
         self.timers._reset_timer('expiration')
         self.state = fsm.State.INIT
         # FIXME: call hooks in a non blocking way (maybe call_soon ?)
-        for i in self.config.hooks:
+        for i in reversed(self.config.hooks):
```

Comment:
Just out of curiosity: why reversed?

Reply:
Sometimes we have to run a hook before another when we get a new lease; for example setting the gateway address depends on having the IP assigned to the interface. But when releasing the lease, we have to remove the gateway before removing the IP or it will fail (because the gateway will already have been implicitly removed when deleting the IP from the interface). That does not actually cause any problems except for spurious errors, but these are better avoided.